### DISCUSSION TITLE
type annotation for link options

### DISCUSSION THREAD

**Discussion by ImSingee** [TanStack/router#2708] [Q&A]
How could I denote a param to accept link options?

For example

```
type MenuItem = {
   // ...
   link: ?
}
```

So that I can use

```
<Link {...props.link} />
```

And one more question, sometimes I want to just accept the `to` (no paths, no queries, etc.), like

```
type MyProps = {
    link: ?
}

<Link to={props.to}>...</Link>
```

Is there any way to do these things? I try to play with `Link` and `linkOptions`, but finally failed

**Comment by CanRau** [comment]
Have you looked into https://tanstack.com/router/latest/docs/framework/react/guide/link-options ?

**Comment by ImSingee** [comment]
@CanRau Thanks for your reply. I have read it, but I can't get how to achieve my needs with it. I've tried `ReturnType<typeof linkOptions>` but it doesn't work
